src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import duke.exception.TaskIndexOutOfBound;
+import duke.exception.EmptyDoneIndexException;
+
+import java.util.ArrayList;
+
+public class TaskManager {
+    public static final int MAX_TASKS = 100;
+    public  ArrayList<Task> tasks = new ArrayList<>();
+
+    public void addTask(Task task){
+        tasks.add(task);
+        System.out.println("Got it. I've added this task: ");
+        System.out.println("  " + task.getTask());
+        System.out.println("Now you have " + tasks.size() + " tasks on the list.");
+    }
+
+    public boolean searchTask(Task task){
+        boolean taskFound = false;
+        for(int i=0; i<tasks.size(); i++){
+            if(tasks.get(i).getTask().equals(task.getTask()) && !tasks.get(i).isDone()){   //be aware that maybe it is added but it is done, so your can add another one
```

Comment:
The logic in the if statement is overly complex making it extremely long and hard to read - better to abstract each part into separate functions